quix-frontend/client/src/lib/runner/services/autocomplete/autocomplete-service.ts

```diff
@@ -38,46 +45,58 @@ export async function setupCompleters(
   const keywordsCompletions = getKeywordsCompletions();
 
   const completerFn = async (prefix: string, session: IEditSession) => {
+    let all;
+    let searchInObject: Boolean;
+
     const { query, position } = getQueryAndCursorPositionFromEditor(
       editorInstance,
       session
     );
+
     const queryContext: QueryContext = evaluateContextFromPosition(
       query,
       position
     );
-    const contextCompletions: Promise<AceCompletion[]> = sqlAutocompleter.getCompletionItemsFromQueryContext(
-      queryContext
-    );
 
-    const [keywords, completions] = await Promise.all([
-      keywordsCompletions,
-      contextCompletions,
-    ]);
-
-    let all =
-      queryContext.contextType === ContextType.Undefined
-        ? keywords
-        : completions;
+    switch (queryContext.contextType) {
+      case ContextType.Undefined:
+        all = keywordsCompletions;
+        break;
+      case ContextType.Table:
+        all = await sqlAutocompleter.getCompletionItemsFromQueryContext(
+          queryContext
+        );
+        break;
+      default:
+        all = await sqlAutocompleter.getCompletionItemsFromQueryContext(
+          queryContext
+        );
+        const filteredCompletions: object[] = all.filter(obj => obj.value.includes(prefix));
+        searchInObject = filteredCompletions.length === 0;
+        if (searchInObject) {
+          all = await sqlAutocompleter.getCompletionItemsFromQueryContextColumn(
+            queryContext
+          );
+        }
+    }
 
     if (prefix) {
       const lowerCasedPrefix = prefix.trim().toLowerCase();
+      if (queryContext.contextType === ContextType.Undefined) {
+        all = enrichCompletionKeyWord(all, lowerCasedPrefix);
```

Comment:
check if i can merge this and 
enrichCompletionItemColumSearch
and give appropriate name
`enrichCompletionItem`

Reply:
combined
and solved filter problem